src/test/ui/macros/recovery-forbidden.rs

```diff
@@ -0,0 +1,12 @@
+// check-pass
+
+macro_rules! dont_recover_here {
+    // The parser recovers to `!1` when it sees `not 1`
+    ($e:expr) => {};
```

Comment:
I should add a `compile_error` to this arm to make it clearer that it should match the other arm.